torch/csrc/distributed/rpc/types.cpp

```diff
@@ -0,0 +1,48 @@
+#include <torch/csrc/distributed/rpc/types.h>
+
+
+namespace torch {
+namespace distributed {
+namespace rpc {
+
+GloballyUniqueId::GloballyUniqueId(worker_id_t createdOn, local_id_t localId)
+    : createdOn_(createdOn), localId_(localId) {}
+
+bool GloballyUniqueId::operator==(const GloballyUniqueId& other) const {
+  return createdOn_ == other.createdOn_ && localId_ == other.localId_;
+}
+
+bool GloballyUniqueId::operator!=(const GloballyUniqueId& other) const {
+  return createdOn_ != other.createdOn_ || localId_ != other.localId_;
+}
+
+at::IValue GloballyUniqueId::toIValue() const {
+  std::vector<at::IValue> ivalues = {(int64_t)createdOn_, (int64_t)localId_};
+  return c10::ivalue::Tuple::create(std::move(ivalues));
+}
+
+GloballyUniqueId GloballyUniqueId::fromIValue(at::IValue&& ivalue) {
+  auto ivalues = ivalue.toTuple()->elements();
```

Comment:
what happens if ivalue.toTuple() conversion fails? Does it throw an exception or does it return a nullptr?

Reply:
It throws an exception if it is not a tuple